src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -858,15 +858,15 @@ public static unsafe bool TryEncode(ReadOnlySpan<char> utf16, Span<byte> utf8, o
                         // if (!IsLowSurrogate(ch) && !IsHighSurrogate(ch))
                         if (!InRange(ch, HIGH_SURROGATE_START, LOW_SURROGATE_END))
                         {
-                            if (pTarget >= pAllocatedBufferEnd - 2)
+                            if (pAllocatedBufferEnd - pTarget <= 2)
```

Comment:
x >= y -2
x - y >= -2
-(x-y) <= 2
-x+y <= 2
y - x <= 2

[x >= y - 2] == [y - x <= 2]
Where is the error?